File: backend/ria_ml/SmithCase_Chapter14.py
import numpy as np
from client_create import client_create
from client_process import client_process
from market_create import market_create
from market_process import market_process
from iSocialSecurity_create import iSocialSecurity_create
from iSocialSecurity_process import iSocialSecurity_process
from analysis_create import analysis_create
from analysis_process import analysis_process

client = client_create()
client = client_process(client)

market = market_create()
market = market_process(market, client)

iSocialSecurity = iSocialSecurity_create()
iSocialSecurity['state1Incomes'] = np.array([[np.inf, 30000]])
iSocialSecurity['state2Incomes'] = np.array([[np.inf, 30000]])
iSocialSecurity['state3Incomes'] = np.array([44000])
client = iSocialSecurity_process(iSocialSecurity, client, market)

analysis = analysis_create()
analysis['plotRecipientPVs'] = 'y'
analysis_process(analysis, client, market)

File: backend/ria_ml/SmithCase_Chapter14.m
% SmithCase_Chapter14.m

% clear all previous variables and close any figures
   clear all;
   close all;
   
% create a new client data structure
   client = client_create();
% process the client data structure 
   client = client_process(client);
 
% create a new market data structure
   market = market_create( );
% process the client data structure
   market = market_process(market,client); 


% create social security 
   iSocialSecurity = iSocialSecurity_create( );
% incomes in state 1, last column repeated for subsequent years
   iSocialSecurity.state1Incomes = [Inf    30000];
% incomes in state 2, last column repeated for subsequent years
   iSocialSecurity.state2Incomes = [Inf    30000];                                   
% incomes for state 3, last column repeated for subsequent years
   iSocialSecurity.state3Incomes = [44000]; 
% process social security
   client = iSocialSecurity_process(iSocialSecurity, client, market); 
   
% create analysis
   analysis = analysis_create( );  
   analysis.plotRecipientPVs = 'y';
% process analysis   
   analysis_process(analysis, client, market);

   
 

  